### PR TITLE
fix: Issues when editing note properties - EXO-73674 - Meeds-io/MIPs#128

### PR DESCRIPTION
**Pull request by hakermi**
Issues when editing note properties

### CODE REVIEW

File: notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java [hunk 1]
```diff
@@ -765,11 +767,28 @@ public Response updateNoteById(@Parameter(description = "Note id", required = tr
...
+        }
+        noteService.createVersionOfNote(note_, identity.getUserId());
+        if (!Utils.ANONYM_IDENTITY.equals(identity.getUserId())) {
+          WikiPageParams noteParams = new WikiPageParams(note_.getWikiType(), note_.getWikiOwner(), newNoteName);
+          noteService.removeDraftOfNote(noteParams, note.getLang());
+        }
+      } else if ((featuredImage != null && (featuredImage.isToDelete() || featuredImage.getUploadId() != null))
+          || !notePageProperties.getSummary().equals(note_.getProperties().getSummary())) {
+        if (StringUtils.isBlank(note.getLang())) {
+          note_.setProperties(notePageProperties);
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java:786]: IMO the properties should be set only when the properties are really modified

File: notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java [hunk 2]
```diff
@@ -765,11 +767,28 @@ public Response updateNoteById(@Parameter(description = "Note id", required = tr
...
           note_ = noteService.updateNote(note_, PageUpdateType.EDIT_PAGE_CONTENT, identity);
         } else {
+          note_.setLang(note.getLang());
           note_ = noteService.updateNote(note_, PageUpdateType.EDIT_PAGE_CONTENT, identity);
           note_.setContent(note.getContent());
+          note_.setProperties(notePageProperties);
+        }
+        noteService.createVersionOfNote(note_, identity.getUserId());
+        if (!Utils.ANONYM_IDENTITY.equals(identity.getUserId())) {
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java:779]: can we put that oustide of if/else ?

File: notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java [hunk 3]
```diff
@@ -208,6 +217,7 @@ public NoteServiceImpl(DataStorage dataStorage,
     this.cmsService = cmsService;
     this.fileService = fileService;
     this.uploadService = uploadService;
+    this.metadataService = metadataService;
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java:225]: we have also ImageThumbnailService thumbnailService = CommonsUtils.getService(ImageThumbnailService.class); used in getNoteFeaturedImageInfo to be modified also

File: notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java [hunk 4]
```diff
@@ -240,24 +250,13 @@ public Page createNote(Wiki noteBook, String parentNoteName, Page note, Identity
...
-              properties.setDraft(false);
-              properties = saveNoteMetadata(properties,
-                                            note.getLang(),
-                                            Long.valueOf(identityManager.getOrCreateUserIdentity(userIdentity.getUserId())
-                                                                        .getId()));
-            }
+            properties.setNoteId(Long.parseLong(createdPage.getId()));
+            properties.setDraft(false);
+            properties =
+                       saveNoteMetadata(properties,
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java]: do we have the case the the featured image is already created by the last draft ?

File: notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java [hunk 5]
```diff
@@ -328,18 +327,24 @@ public Page updateNote(Page note, PageUpdateType type, Identity userIdentity) th
...
-    if (notePageProperties != null) {
-      notePageProperties.setDraft(false);
-      org.exoplatform.social.core.identity.model.Identity identity =
-                                                                   identityManager.getOrCreateIdentity(OrganizationIdentityProvider.NAME,
-                                                                                                       userIdentity.getUserId());
-      saveNoteMetadata(note.getProperties(), note.getLang(), Long.valueOf(identity.getId()));
+    if (userIdentity != null) {
+      try {
+        NotePageProperties properties = note.getProperties();
+        properties.setNoteId(Long.parseLong(updatedPage.getId()));
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java:344]: do we need to test if (notePageProperties != null) ?

**Review by azayati — CHANGES_REQUESTED** [review]

File: notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java (hunk 1 above)
Comment (hakermi) [reply notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java:786]: and what this condition means ? :  `else if ((featuredImage != null && (featuredImage.isToDelete() || featuredImage.getUploadId() != null))
          || !notePageProperties.getSummary().equals(note_.getProperties().getSummary()))`

File: notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java (hunk 2 above)
Comment (hakermi) [reply notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java:779]: No, we can't

File: notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java (hunk 4 above)
Comment (hakermi) [reply notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java]: already handled in the `saveMetadata` function

File: notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java (hunk 5 above)
Comment (hakermi) [reply notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java:344]: Yes, Done

File: notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java (hunk 3 above)
Comment (hakermi) [reply notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java:225]: Done

**Review by azayati — APPROVED** [review]
LGTM
